control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1073,20 +1073,30 @@ func (r *HostedControlPlaneReconciler) ensureControlPlane(ctx context.Context, h
 	r.Log.Info("successfully applied all manifests")
 
 	var kubeadminPassword string
-	kubeadminPasswordSecret := common.KubeadminPasswordSecret(targetNamespace)
-	if _, err := r.CreateOrUpdate(ctx, r, kubeadminPasswordSecret, func() error {
-		return reconcileKubeadminPasswordSecret(kubeadminPasswordSecret, hcp, &kubeadminPassword)
-	}); err != nil {
-		return fmt.Errorf("failed to reconcile kubeadminPasswordSecret: %w", err)
+	if hcp.Status.KubeadminPassword != nil {
```

Comment:
here after that change is made on the hostedcluster you would use that to decide if you need to reconcile or not.....